src/code-converter.ts

```diff
@@ -226,4 +233,34 @@ export class CodeConverter {
         }
         return mqttPublishString;
     }
+
+    private identifyDatapoints(finalizedMainPy: string): DataPointDefinition[] {
+        const finalizedMainPyArray = createArrayFromMultilineString(finalizedMainPy);
+        const dataPointsMap = new Map();
+        const dataPoints: DataPointDefinition[] = [];
+        finalizedMainPyArray.forEach((line: string) => {
+            if (line.includes('.Vehicle.')) {
+                const captureAlternatives = '\\.subscribe|\\.get|\\.set|\\)';
+                const dataPointRegExp = new RegExp(`Vehicle.*?(${captureAlternatives})`);
+                const dataPointMatch = dataPointRegExp.exec(line);
+                if (dataPointMatch) {
+                    const dataPointPath = dataPointMatch[0].split(dataPointMatch[1])[0];
+                    switch (dataPointMatch[1]) {
+                        case '.set':
+                            dataPointsMap.set(dataPointPath, 'write');
+                            break;
+                        default:
+                            if (!dataPointsMap.has(dataPointPath)) {
+                                dataPointsMap.set(dataPointPath, 'read');
+                            }
+                            break;
+                    }
+                }
+            }
+        });
+        dataPointsMap.forEach((dataPointAccess: string, dataPointPath: string) =>
```

Comment:
Are these parameters reversed? Path is the key of your map, access is the value of your map.

Reply:
The callback function of the map.forEach starts with value instead of key :<

Map<any, any>.forEach(callbackfn: (value: any, key: any, map: Map<any, any>)

Reply:
OK